### PR TITLE
Adding support for Golang coverage generated using `-coverpkg` flag

### PR DESCRIPTION
**Pull request by safety-jure**
## Issue Summary
When running Golang tests with `-coverpkg` flag we are instruction Go to measure total coverage against packages listed. This would include cross-package coverage. I.e. tests from package A that implicitly test package B would contribute towards total coverage.

Example command:
```
go test \
  -coverpkg="github.com/codeclimate/test-reporter/formatters/gocov/example/foo,github.com/codeclimate/test-reporter/formatters/gocov/example/bar" \
  -coverprofile=c.out \
  ./...
```

Problem with this is that source files might appear more than once in the generated report and current code do not handle that.

## Changes
• Updated `gocov` formatter, with an extra step where we de-duplicate blocks already covered.
• Added a unit test for Golang coverage report generated using `-coverpkg` flag

### CODE REVIEW

**Comment by brendanlucas** [comment]
Please note, this PR is a fix for support request #40191

Thanks 🙂

File: formatters/gocov/example_foobar.out [hunk 1]
```diff
@@ -0,0 +1,22 @@
+mode: set
```
Comment (fede-moya) [inline formatters/gocov/example_foobar.out:1]: @safety-jure could you move this file to `formatters/gocov/example/` ?

**Review by fede-moya — CHANGES_REQUESTED** [review]
Makes perfect sense, thanks for the description and your contribution. Approve after moving the requested file. 👍🏼

File: formatters/gocov/example_foobar.out (hunk 1 above)
Comment (safety-jure) [reply formatters/gocov/example_foobar.out]: Done. However, the Linux job is failing now for an unrelated reason. Let me know @fede-moya if there is anything else I should do. 

Thank you!

Comment (fede-moya) [reply formatters/gocov/example_foobar.out]: Great. I have just opened a PR to fix ci. Merging this one after we solve the ci issue, will ask you to rebase later.  👍🏼

**Comment by fede-moya** [comment]
Hey @safety-jure can you rebase master ? I have just merge a fix for the CI.

**Comment by safety-jure** [comment]
Thank you @fede-moya 

I just rebased master and `linux` job successfully completed, however, now the macos one is failing, for the same reason. 

I reckon we also need to change line 30 in `.circleci/config.yml ` to 
```
      - run: brew install go@1.15
```

**Comment by fede-moya** [comment]
@safety-jure Feel free to rebase again 👍🏼

**Comment by safety-jure** [comment]
> @safety-jure Feel free to rebase again 👍🏼

Done and it is looking good now. Thanks @fede-moya

**Review by fede-moya — APPROVED** [review]

**Comment by fede-moya** [comment]
Hey @safety-jure had the binaries worked for you ?

**Comment by fede-moya** [comment]
Hey @safety-jure , have you been able to try out the latest binaries ?

**Pull request merged**